### PR TITLE
options and terminology fixes and docs added to extension code

### PR DESCRIPTION
**Pull request by brendankenny**
cleanup of extension code, which has grown a little hairy as it has accumulated. Mostly:
- always start with a freshly parsed config.json when running lighthouse
- finish removal of audit whitelist in #608 by filtering audits in config directly before passing into LH, but make selected audit categories work again.
- `s/audits/aggregations` after #595, to reflect that it's aggregation categories being saved and loaded for the options list, not audits themselves
- added some docs so it's hopefully a little faster to find your way around in there

This PR fixes 2 bugs:

1. the audit whitelist was still being passed into the `Config` constructor. Amazingly, the browserify polyfill for `path.dirname` parses `new Set(['array', 'of', 'audit', 'names'])` as `'.'`, so this wasn't actually causing any problems.
1. Currently on master the options page has no affect on the audits run. This makes that work again

### CODE REVIEW

**Comment by brendankenny** [comment]
> Admittedly the only actual bug being fixed here

Actually, that's not true. Currently on master the options page has no affect on the audits run. This makes that work again

File: lighthouse-extension/app/src/lighthouse-background.js [hunk 1]
```diff
@@ -14,14 +14,16 @@
...
 'use strict';
 
 const ExtensionProtocol = require('../../../lighthouse-core/gather/drivers/extension');
 const Runner = require('../../../lighthouse-core/runner');
 const Config = require('../../../lighthouse-core/config/config');
-const configJSON = require('../../../lighthouse-core/config/default.json');
+const defaultConfig = require('../../../lighthouse-core/config/default.json');
 const log = require('../../../lighthouse-core/lib/log');
+
+const defaultConfigJSON = JSON.stringify(defaultConfig);
```
Comment (paulirish) [inline lighthouse-extension/app/src/lighthouse-background.js]: lets do this stringify down where we parse. too many configs running about :)

Comment (brendankenny) [inline lighthouse-extension/app/src/lighthouse-background.js]: done

File: lighthouse-extension/app/src/popup.js [hunk 2]
```diff
@@ -47,72 +48,88 @@ document.addEventListener('DOMContentLoaded', _ => {
...
-    .then(audits => background.runAudits({
-      flags: {
-        mobile: true,
-        loadPage: true
-      }
-    }, audits))
+    background.loadSelectedAggregations()
+    .then(getAuditsFromSelected)
+    .then(selectedAudits => {
+      return background.runLighthouse({
```
Comment (wardpeet) [inline lighthouse-extension/app/src/popup.js]: @brendankenny why use a return here?

Comment (brendankenny) [inline lighthouse-extension/app/src/popup.js]: just a (debatable :) readability improvement. It's often easy to miss that a multiline body on an arrow function is being returned

**Pull request merged**